samples/tls_between_enclaves/AttestedTLSREADME.md

```diff
@@ -0,0 +1,153 @@
+# Secure Channel in OE SDK
+
+  As Open Enclave SDK getting adopted into more realistic scenarios, we are receiving requests from OE SDK developers for adding secure channel support.
+
+  We do have the remote_attestation sample that shows how to conduct remote attestation between two enclaves and establish a `proprietary` channel based on asymmetric keys exchanged during the attestation process. It demonstrates how to conduct mutual attestation but it does not go all the way to show how to establish a fully secure channel.
+
+  Most of the real world software uses TLS-like standard protocol through popular TLS APIs (OpenSSL, WolfSSL, Mbedtls...) for establishing secure channels. Thus, instead of inventing a new communication protocol, we implemented `Attested TLS` feature to address above customer need by adding a set of new OE SDK APIs to help seamlessly integrate remote attestation into the popular TLS protocol for establishing an TLS channel with attested connecting party without modifying existing TLS APIs (such as OpenSSL, Mbedtls, and others).
+
+# What is an Attested TLS channel
+
+The remote attestation feature that comes with TEE (such as Intel SGX or ARM's TrustZone enclave, in the context of this doc) could significantly improve a TLS endpoint (client or server) trustworthiness for a TLS connection starting or terminating inside an enclave. An Attested TLS channel is a TLS channel that integrates remote attestation validation as part of the TLS channel establishing process. Once established, it guarantees that an attested connecting party is running inside a TEE with expected identity.
+
+There are two types of Attested TLS connections:
+1. Both ends of an Attested TLS channel terminate inside TEE
+    - Guarantee that both parties of a TLS channel are running inside trustes TEEs
+    - OE SDK sample: tls_between_enclave
+2. Only one end of an Attested TLS channel terminate inside TEE
+    - In this case, the assumption is that the end not terminated inside an TEE is a trust party. The most common use case is, this non-Tee party might have secrets to securely share with the other party through an Attested TLS channel.
+    - OE SDK sample: tls_between_host_enclave
+
+## Prerequisites
+
+  The audience is assumed to be familiar:
+
+  - [Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security) a cryptographic protocol designed to provide communications security over a computer network.
+
+  - [Open Enclave Remote Attestation](https://github.com/Microsoft/openenclave/tree/master/samples/remote_attestation#what-is-attestation): Remote Attestation is the concept of a HW entity or of a
+combination of HW and SW gaining the trust of a remote provider or producer.
+
+### How it works
+
```

Comment:
"we make this certificate" instead of "In this certificate, we make"

Reply:
fixed